src/Module/Contact.php

```diff
@@ -206,15 +201,26 @@ private static function updateContactFromProbe(int $contact_id)
 	}
 
 	/**
-	 * Toggles the blocked status of a contact identified by id.
+	 * Toggles the blocked status of a contact identified by a public or private id.
+	 * Also dissolves any existing relationship when a contact is blocked.
 	 *
-	 * @param int $contact_id Id of the contact with uid = 0
-	 * @throws \Exception
+	 * @param array $cdata    Result of Model\Contact::getPublicAndUserContactID for the target contact
+	 * @param int   $owner_id Id of the user we want to block the contact for
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws \ImagickException
 	 */
-	private static function toggleBlockContact(int $contact_id)
+	private static function toggleBlockContact(array $cdata, int $owner_id)
 	{
-		$blocked = !Model\Contact\User::isBlocked($contact_id, local_user());
-		Model\Contact\User::setBlocked($contact_id, local_user(), $blocked);
+		$newBlockedState = !Model\Contact\User::isBlocked($cdata['public'], local_user());
+		if ($newBlockedState && $cdata['user']) {
+			$contact = Model\Contact::getContactForUser($cdata['user'], $owner_id);
+			$owner = Model\User::getOwnerDataById($owner_id);
+			if (DBA::isResult($contact) && DBA::isResult($owner)) {
+				Model\Contact::terminateFriendship($owner, $contact, true);
```

Comment:
Terminating the connection when blocking someone is a problem since the connection isn't restored when the connection is unblocked.

Reply:
This is the standard behavior on both Facebook and Twitter, I don't see a need to restore the connection on unblocking. The Friendica user will be able to follow them again on their own, which is the important part we have control over.

Reply:
What happens when you accidentally block someone? You won't realize that you unfollowed that contact. You will revert the block and think that everything went well. And even some message won't help because people don't read messages.

Reply:
I'm guessing that "blocking" has a very cold, technical meaning for you that it usually doesn't carry for most other people. Blocking is the nuclear option, it's the be-all and end-all of online relationship breakup, and as such it is expected to leave no trace of any former relationship. On Twitter for example, it is known as the only way to revoke a follow or to be forcibly removed from a contact list created by someone else.

When it is done by mistake, it is a big deal and people will realize that they aren't following that contact, if only because we will say as much on the contact page.

Reply:
I don't think that people will always know if the other side followed you as well. And even then when you wanted to revert the blocking you had not only to follow that person again but had to write the other person to follow you again because you accidentally blocked that person.

I think that the user should decide on their own whether to block, unfollow, or revert the subscription of a contact.